ext/console/02_console.js

```diff
@@ -739,6 +739,40 @@
     );
   }
 
+  // Multi byte character aware string truncation
+  // Taken from https://github.com/lautis/unicode-substring/blob/master/index.js
+  // MIT License. Thanks @lautis.
+  function multibyteAwareTruncate(string, end) {
+    function charAt(string, index) {
+      const first = string.charCodeAt(index);
+      let second;
+      if (first >= 0xD800 && first <= 0xDBFF && string.length > index + 1) {
+        second = string.charCodeAt(index + 1);
+        if (second >= 0xDC00 && second <= 0xDFFF) {
+          return string.substring(index, index + 2);
+        }
+      }
+      return string[index];
+    }
+
+    let accumulator = "";
+    let character;
+    let stringIndex = 0;
+    let unicodeIndex = 0;
+    const length = string.length;
+
+    while (stringIndex < length) {
+      character = charAt(string, stringIndex);
+      if (unicodeIndex < end) {
+        accumulator += character;
+      }
+      stringIndex += character.length;
+      unicodeIndex += 1;
+    }
+
+    return accumulator;
+  }
+
```

Comment:
We don't really care about what non-BMP or lone surrogate characters there were at any point before where we're truncating the string. So we could just check whether `end - 1` is a lead surrogate (0xD800 to 0xDBFF), and if it is, truncate at `end - 1`. This would do the "wrong" thing if there's a lone lead surrogate at that point, but I don't think that's a problem.

But in either case, whether in this implementation or the way I suggested, it is possible for the truncated string to end with a lead surrogate (if there's a lone lead surrogate at index 99, a lead surrogate at 100, and a trail surrogate at 101).

Reply:
@andreubotella Sounds reasonable. I'll update the PR.

Reply:
Updated now.